docs/acceptance-criteria/uacs-rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,467 @@
+  pool to the submitter’s address (in the same tx) +
+// TODO: revisit 2 weeks removal time
+  - operators who became eligible for submitting the DKG result before the
+  member who submitted the DKG result should be punished by being removed from
+  the sortition pool for a fixed amount of time (2 weeks) (in the same tx).
+
+| Group creation - DKG timeout
+| The DKG submission timeout equals the group size multiplied by the number of
+  blocks for a member to become eligible to submit the DKG result. The timer
+  starts at the moment when the first member becomes eligible (the moment when
+  DKG was requested). The timer gets reset when a valid DKG result challenge is
+  submitted. 
+
+| Group creation - DKG timeout
+| Anyone can send a transaction reporting DKG timeout.
+
+| Group creation - DKG timeout
+| Upon noticing the DKG timeout notification, the random beacon contract should
+  determine its validity.
+
+| Group creation - DKG timeout
+| When determining validity of the DKG timeout notification, the following
+  aspects should be taken into account: +
+  - does it reference the existing DKG request? (if not, then notification
+  invalid) +
+  - has the DKG timeout passed? (if not, then notification invalid) +
+  - is the sortition pool in a locked state (if not - meaning somebody already
+  unlocked it - then notification invalid).
+
+| Group creation - DKG timeout
+| If the DKG timeout notification was deemed justified, then: +
+  - the pool should be unlocked +
+  - the fixed amount reward (in T) should be sent from the maintenance pool to
+  the notifier.
+
+| Sortition pool - punishments
+// TODO: revisit 2 weeks removal time
+| Member of the sortition pool can be punished with 2 weeks removal from the
+  pool as a result of: +
+  - misbehavior during the DKG submission +
+  - misbehavior during relay entry submission.
+
+| Sortition pool - punishments
+| Operators punished with removal from the group should not earn rewards for the
+  time of their absence in the pool.
+
+| Sortition pool - punishments
+// TODO: revisit 2 weeks removal time
+| After 2 weeks punish period elapses, operators removed from the sortition pool
+  should be automatically reinstated in the pool and should be able to earn the
+  rewards again (from membership in the pool and potentially submission of the
+  DKG result or unlocking the pool).
+
+| Sortition pool - punishments
+| The operation removing members from the sortition pool should be as cheap as
+  possible so that the member submitting the relay entry or the DKG result does
+  not have to pay any additional costs for removing inactive members from the
+  pool.
+
+| Sortition pool - rewards
+| Operators in the sortition pool can earn T for multiple activities: +
+  - for staying in the pool (given weekly, proportionally to stake) +
+  - for successful submission of the DKG result (fixed value) +
+  - for successful unlocking of the sortition pool if DKG timed out.
+
+| Sortition pool - locking
```

Comment:
I think we should also mention that during locking, operators should not be able to update their state. This is because they can be removed during the update.

Reply:
Added, see e5d9a5989baaf4a860d0007855a1556af7d1781d.